static/src/javascripts/projects/common/modules/identity/email-sign-in-banner/index.js

```diff
@@ -50,21 +40,44 @@ const canShow: () => Promise<boolean> = () => {
             isInExperiment() &&
             isSecondEmailPageview()
     );
+    return can;
+};
+
+const sideEffects: () => void = () => {
     const cmp = getEmailCampaignFromUrl();
-    if (cmp) {
-        userPrefs.set(userPrefsReferrerKey, cmp.utm);
+    const existing = userPrefs.get(userPrefsStoreKey);
+    if (cmp && !existing) {
+        userPrefs.set(userPrefsStoreKey, {
+            utm: cmp.utm,
+            pv: 1,
+        });
+    } else if (existing && cmp && existing.utm !== cmp.utm) {
```

Comment:
So the intention of this logic is that both these example cases would trigger a banner?

1. User clicks on email link, then navigates to a second article using a standard onward journey.
2. User clicks on email link, closes browser, comes back later to same email and clicks another link.

Reply:
nope, good shout!

it should only be on a direct second page journey, if they navigate out then back in to page 2 they shouldn't see it in page 1

Reply:
update: this will take place in a different check (there are a couple more of them planned, such as display only once per week for example)